assets/js/copy-code-blocks.js

```diff
@@ -0,0 +1,40 @@
+// copy-code-blocks.js
+// Copyright: 2023  Scientific Python Developers
+// Author: Adam Porter <adam@alphapapa.net>
+// License: BSD 3-Clause License
+//   see <https://raw.githubusercontent.com/scientific-python/scientific-python-hugo-theme/main/LICENSE>
+
+// This file implements a button attached to code blocks which copies
+// the code to the clipboard.
+
+// This selector should match the PRE elements, each of which may
+// itself contain the text to be copied, or may contain a CODE element
+// which contains the text to be copied.
+const pre_selector = "div.highlight pre";
+
+document.querySelectorAll(pre_selector).forEach(add_button_to);
+
+function add_button_to(element) {
+    button = document.createElement("button");
+    button.classList.add("copy-button")
+    button.append(document.createTextNode("Copy"));
+    button.addEventListener("click", copy_content_of);
+    element.prepend(button);
+}
+
+function copy_content_of(event) {
+    // Omit "Copy" prefix from the button itself.
+    target = this.parentElement.textContent.substring(4);
```

Comment:
What is the substring(4) used for?

Reply:
It was sort of an elegant hack, but no longer needed after other changes.